apps/omg_watcher/lib/omg_watcher/db/txoutput.ex

```diff
@@ -67,18 +76,34 @@ defmodule OMG.Watcher.DB.TxOutput do
     # TODO: Make use of Block API's block.get when available
     do: DB.Transaction.get_by_blknum(blknum) |> Core.compose_output_exit(decoded_utxo_pos)
 
+  # preload ethevents in a single query as there will not be a large number of them
   @spec get_by_position(Utxo.Position.t()) :: map() | nil
   def get_by_position(Utxo.position(blknum, txindex, oindex)) do
-    Repo.get_by(__MODULE__, blknum: blknum, txindex: txindex, oindex: oindex)
+    DB.Repo.one(
+      from(txoutput in __MODULE__,
+        preload: [:ethevents],
+        left_join: ethevent in assoc(txoutput, :ethevents),
+        where: txoutput.blknum == ^blknum and txoutput.txindex == ^txindex and txoutput.oindex == ^oindex
+      )
+    )
   end
 
   def get_utxos(owner) do
     query =
       from(
-        txo in __MODULE__,
-        where: txo.owner == ^owner and is_nil(txo.spending_txhash) and is_nil(txo.spending_exit),
-        order_by: [asc: :blknum, asc: :txindex, asc: :oindex],
-        preload: [:creating_transaction, :deposit]
+        txoutput in __MODULE__,
+        preload: [:ethevents],
+        left_join: ethevent in assoc(txoutput, :ethevents),
+        where:
+          txoutput.owner == ^owner and
+            is_nil(txoutput.spending_txhash) and (is_nil(ethevent) or fragment("
```

Comment:
some short, verbal summary of what the SQL fragment means would be welcome. Or alternatively DRY both occurrences to some named function that would clarify what's up?

Reply:
totally agreed. i've tried to pull out the fragment, but ecto doesn't like it. 😢 I will be working on this code a lot more as I need to add in-flight exits, challenges, etc... let me see if I can get this DRY later. I've added comment for now.